streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -194,13 +197,14 @@ public synchronized void close() {
                     Thread.interrupted();
                 }
             }
+        }
 
+        if (state != STOPPED) {
+            metrics.close();
```

Comment:
Moved `metrics.close()` to a new conditional. This should only be able to run once. Once state goes to `STOPPED`, `close` is a no-op and `start` should throw an error.